RedHat-UX/red-hat-design-system
# docs: fix all patterns page layout + content issues

## What I did

1. Fixed some [`z-index` issues](https://github.com/user-attachments/assets/b9ca1b60-9e2e-4ca0-8668-0ba79c7a7186) on the All Patterns page related to `<rh-tile>`, `<uxdot-masthead>`, and `<uxdot-sidenav>`.
2. Moved the All Patterns descriptions into the default slot
3. Added descriptions for Tabs and Tile
4. Added / updated thumbnail images 

## Testing Instructions

1. Check out the [All Patterns page on uxdot](https://ux.redhat.com/patterns/). Be sure to scroll vertically and view it on mobile/tablet with the menu open.
2. Now view the All Patterns page in [this PR's DP](https://deploy-preview-1916--red-hat-design-system.netlify.app/patterns/).
3. Ensure there are no `z-index` issues.
4. Ensure the imagery, titles and descriptions are accurate and grammatically correct.
5. Ensure the sizing for the descriptions is correct.

## Notes to Reviewers

Strangely, on `localhost:8080`, the tile for Logo Wall appears first. Then, in the DP, the Logo Wall tile is correctly ordered. 🤔  Ten brownie points go to whoever can explain that.
